overlord/snapstate/snapstate.go

```diff
@@ -94,7 +94,8 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 			return nil, err
 		}
 		if model == nil || model.Base() == "" {
-			if !experimentalAllowSnapd {
+			// always allow the snapd snap on classic
+			if !release.OnClassic && !experimentalAllowSnapd {
```

Comment:
Shouldn't experimentalAllowSnapd == false lead to a different error? Seems to me that having same error message for both cases may be confusing.

Reply:
I reworked this code a little bit to make it easier to read. I think there is only a single error condition (unless I missread the comment) but the code was a bit confusing, I hope the new version is easier to read.